hadoop-hdds/tools/src/main/java/org/apache/hadoop/hdds/scm/cli/pipeline/ListPipelinesSubcommand.java

```diff
@@ -54,17 +54,12 @@
   @Override
   public Void call() throws Exception {
     try (ScmClient scmClient = parent.getParent().createScmClient()) {
-      if (Strings.isNullOrEmpty(factor) && Strings.isNullOrEmpty(state)) {
-        scmClient.listPipelines().forEach(System.out::println);
-      } else {
-        scmClient.listPipelines().stream()
-            .filter(p -> ((Strings.isNullOrEmpty(factor) ||
-                (p.getFactor().toString().compareToIgnoreCase(factor) == 0))
-                && (Strings.isNullOrEmpty(state) ||
-                (p.getPipelineState().toString().compareToIgnoreCase(state)
-                    == 0))))
-            .forEach(System.out::println);
-      }
+      scmClient.listPipelines().stream()
+          .filter(p -> ((Strings.isNullOrEmpty(factor)
+              || (p.getFactor().toString().compareToIgnoreCase(factor) == 0))
+              && (Strings.isNullOrEmpty(state) || (p.getPipelineState()
+                  .toString().compareToIgnoreCase(state) == 0))))
+          .forEach(System.out::println);
```

Comment:
Structuring this as below would reduce duplication, and make it easier to add further filters if needed:

```
Stream<...> stream = scmClient.listPipelines().stream();
if (!isNullOrEmpty(factor)) {
  stream = stream.filter(...);
}
if (!isNullOrEmpty(state)) {
  stream = stream.filter(...);
}
stream.forEach(...);
```

Reply:
@bhemanthkumar i think the changes within the try block @adoroszlai is
```java
      Stream<Pipeline> stream = scmClient.listPipelines().stream();
      if (!Strings.isNullOrEmpty(factor)) {
        stream =
            stream.filter(p -> p.getFactor().toString()
                .compareToIgnoreCase(factor) == 0);
      }
      if (!Strings.isNullOrEmpty(state)) {
        stream =
            stream.filter(p -> p.getPipelineState().toString()
                .compareToIgnoreCase(state) == 0);
      }
      stream.forEach(System.out::println);
```